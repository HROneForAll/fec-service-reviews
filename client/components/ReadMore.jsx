import React from 'react';
import Review from './Review.jsx';
import style from './style.css.js';

const ReadMore = (props) => {
  var numberOfReviews = props.reviews.length;
  if (props.showMoreButton) {
    return (
      <a href='#' style={style.showReviews} onClick={props.handleSwitch}>Read all {numberOfReviews} reviews</a>
    )
  } else {
    return (
      <a href='#' style={style.showReviews} onClick={props.handleSwitch}>Hide all reviews</a>
    )
  }
};
export default ReadMore;