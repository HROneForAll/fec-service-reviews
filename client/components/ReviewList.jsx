import React from 'react';
import SingleReview from './SingleReview.jsx';
import ReviewModal from './ReviewModal.jsx';
import style from './style.css.js';
const ReviewList = (props) => {
  var reviews = props.reviews;
  var reviewList = [];
  if (props.reviews.length !== 0) {
    for (var i = 0; i < 6; i++) {
      reviewList.push(<SingleReview review={reviews[i]} key={i} showModal={props.showModal} />);
    }
  }
  if (props.showModal) {
    return (
      <div>
        <ReviewModal showModal={props.showModal} />
      </div>
    )
  } else {
    return (
      <div style={style.reviewList}>
        {reviewList}
      </div>
    );
  }
};
export default ReviewList;