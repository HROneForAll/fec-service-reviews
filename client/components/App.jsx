import React from 'react';
import axios from 'axios';
import ReviewList from './ReviewList.jsx';
import ReadMore from './ReadMore.jsx';
import style from './style.css.js';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      reviews: [],
      showModal: false,
      showMoreButton: true,
    };
    this.handleSwitch = this.handleSwitch.bind(this);
  }

  getReviews() {
    axios.get('/1/reviews')
      .then((data) => {
        this.setState({ reviews: data.data });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  handleSwitch() {
    this.setState({
      showModal: !this.state.showModal,
      showMoreButton: !this.state.showMoreButton
    })
  }

  componentDidMount() {
    this.getReviews();
  }

  render() {
    return (
      <div style={style.wrapper} >
        <div style={style.reviewsTitle}>
          Reviews
        </div>
        <ReviewList reviews={this.state.reviews} showModal={this.state.showModal} />
        <ReadMore showMoreButton={this.state.showMoreButton} handleSwitch={this.handleSwitch} />
      </div>
    );
  }
}

export default App;




