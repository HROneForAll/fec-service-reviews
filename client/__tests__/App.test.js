import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App.jsx';
import ReviewList from '../components/ReviewList';
import ReadMore from '../components/ReadMore';
describe('App', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);

    expect(component).toMatchSnapshot();
  });
  it('should render correctly with no props', () => {
    const component = shallow(<App />);

    expect(component).toMatchSnapshot();
  });

});

describe('ReviewList tests', () => {

  it('renders reviewList', () => {
    const items = ['one', 'two', 'three'];
    const wrapper = shallow(<ReviewList reviews={items} />);
    expect(wrapper.find('.reviewList')).toBeDefined();
    // expect(wrapper.find('.item')).toHaveLength(items.length);
  });
});

describe('ReadMore tests', () => {

  it('renders readMore', () => {
    const items = ['one', 'two', 'three'];
    const wrapper = shallow(<ReadMore reviews={items} />);
    expect(wrapper.find('.readMore')).toBeDefined();
    // expect(wrapper.find('.item')).toHaveLength(items.length);
  });
});