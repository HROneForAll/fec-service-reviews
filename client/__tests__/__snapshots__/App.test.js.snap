// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<div
  style={
    Object {
      "color": "#484848",
      "fontFamily": "Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important",
      "marginLeft": "2%",
      "padding": "1%",
    }
  }
>
  <div
    style={
      Object {
        "color": "#484848",
        "fontFamily": "Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif",
        "fontSize": "22px",
        "fontWeight": "600",
        "marginLeft": "2%",
        "wordWrap": "break-word",
      }
    }
  >
    Reviews
  </div>
  <ReviewList
    className="reviewList"
    reviews={Array []}
    showModal={false}
  />
  <ReadMore
    handleSwitch={[Function]}
    showMoreButton={true}
  />
</div>
`;

exports[`App should render correctly with no props 1`] = `
<div
  style={
    Object {
      "color": "#484848",
      "fontFamily": "Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important",
      "marginLeft": "2%",
      "padding": "1%",
    }
  }
>
  <div
    style={
      Object {
        "color": "#484848",
        "fontFamily": "Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif",
        "fontSize": "22px",
        "fontWeight": "600",
        "marginLeft": "2%",
        "wordWrap": "break-word",
      }
    }
  >
    Reviews
  </div>
  <ReviewList
    className="reviewList"
    reviews={Array []}
    showModal={false}
  />
  <ReadMore
    handleSwitch={[Function]}
    showMoreButton={true}
  />
</div>
`;
